uvtools/dspec.py

```diff
@@ -60,11 +60,356 @@ def calc_width(filter_size, real_delta, nsamples):
+                        'clean', iterative clean
+                        'dpss_lsq', dpss fitting using scipy.optimize.lsq_linear
+                        'dft_lsq', dft fitting using scipy.optimize.lsq_linear
+                        'dpss_matrix', dpss fitting using direct lin-lsq matrix
+                                       computation. Slower then lsq but provides linear
+                                       operator that can be used to propagate
+                                       statistics and the matrix is cached so
+                                       on average, can be faster for data with
+                                       many similar flagging patterns.
+                        'dft_matrix', dft fitting using direct lin-lsq matrix
+                                      computation. Slower then lsq but provides
+                                      linear operator that can be used to propagate
+                                      statistics and the matrix is cached so
+                                      on average, can be faster for data with
+                                      many similar flagging patterns.
+                                      !!!WARNING: In my experience,
+                                      'dft_matrix' option is numerical unstable.!!!
+                                      'dpss_matrix' works much better.
+                        'dayenu', apply dayenu filter to data. Does not
+                                 deconvolve subtracted foregrounds.
+                        'dayenu_dft_leastsq', apply dayenu filter to data
+                                 and deconvolve subtracted foregrounds using
+                                'dft_leastsq' method (see above).
+                        'dayenu_dpss_leastsq', apply dayenu filter to data
+                                 and deconvolve subtracted foregrounds using
+                                 'dpss_leastsq' method (see above)
+                        'dayenu_dft_matrix', apply dayenu filter to data
+                                 and deconvolve subtracted foregrounds using
+                                'dft_matrix' mode (see above).
+                                !!!WARNING: dft_matrix mode is often numerically
+                                unstable. I don't recommend it!
+                        'dayenu_dpss_matrix', apply dayenu filter to data
+                                 and deconvolve subtracted foregrounds using
+                                 'dpss_matrix' method (see above)
+                        'dayenu_clean', apply dayenu filter to data. Deconvolve
+                                 subtracted foregrounds with 'clean'.
+                    filter2d: bool
+                        specify whether filtering will be performed in 2d or 1d.
+                        If filter is 1d, it will be applied across the -1 axis.
+                    fitting_options: dict
```

Comment:
so the problem here, is that people who are using clean don't know what the default parameters are, b/c they aren't in the function signature. can't fix that given how you've implemented it here, but just FYI

Reply:
That is true. I will indicate defaults in the docstring.